File: packages/gunreip/laravel-translation-workbench/resources/views/pages/tw-graph/samples/documentation/idea-to-paper/flow/while/code-examples/while-action-sequence.php
<?php
$groups = loadGroups();
$groupIndex = 0;
while ($groupIndex < count($groups)) {
    $items = $groups[$groupIndex];
    prepareGroup($items);
    $itemIndex = 0;
    while ($itemIndex < count($items)) {
        processItem($items[$itemIndex]);
        $itemIndex = $itemIndex + 1;
    }
    finalizeGroup($items);
    $groupIndex = $groupIndex + 1;
}
showSummary();

File: packages/gunreip/laravel-translation-workbench/resources/views/pages/tw-graph/samples/documentation/idea-to-paper/flow/while/code-examples/while-action-sequence.cpp
auto groups = loadGroups(); // std::vector<std::vector<Item>>
std::size_t groupIndex = 0;
while (groupIndex < groups.size()) {
    const auto& items = groups[groupIndex];
    prepareGroup(items);
    std::size_t itemIndex = 0;
    while (itemIndex < items.size()) {
        processItem(items[itemIndex]);
        itemIndex = itemIndex + 1;
    }
    finalizeGroup(items);
    groupIndex = groupIndex + 1;
}
showSummary();
